### ISSUE TITLE
Don't remeber screen orientation

### ISSUE THREAD

**Issue by SilverGreen93** [koreader/koreader#5772]
* KOReader version: 2019.12
* Device: Kobo Forma

#### Issue
When reading from the device in a specific orientation, let's say portrait with hardware buttons on the left, then switch to main file browser, rotate the screen in portrait with harware buttons to the right and open the previous book again, the book automatically rotates the screen back 180 degrees as it remembered the orientation from when it was closed.
This is a pain to rotate back. You have to rotate the device again 180 degrees, wait, and then another 180 degrees to register.
My suggestion here would be NOT to remember the previous rotation of the book and always open the book in the current screen orientation.

#### Steps to reproduce
Open book, rotate screen. Close book, rotate screen. Open book again, screen auto-rotates as it was previously rotated instead of opening in the current screen orientation.

**Comment by pazos** [comment]
> My suggestion here would be NOT to remember the previous rotation of the book and always open the book in the current screen orientation.

Please don't do that. I'm a hardcore user of this behaviour.

> Steps to reproduce

But if you open your document from filemanager then the current rotation will always be portrait.

My two cents:

Add an option somewhere to skip storing / updating metadata with current rotation values. Leaving the default `    ["screen_mode"] = "portrait",`

**Comment by NiLuJe** [comment]
Stupid question: If you actually read in lefty mode (buttons on the right), why would you put them back on the right in the FM? (Because the FM always spawns in Portrait, maybe? I honestly can't remember ^^).

That said, Inverted Portrait is the only non-standard orientation that the FM supports (and that only via gyro events), so that wouldn't help for Landscape readers.
And makes your suggestion unusable anyway, until someone works on making the FM Landscape-friendly ;).

TL;DR: There's no good answer. What's currently happening and what @poire-z are the best that can be done without too much work.

(And, as I read neither in Landcsape nor in Inverted Portrait, I don't really plan on working on that ;)).

**Comment by pazos** [comment]
> (And, as I read neither in Landcsape nor in Inverted Portrait, I don't really plan on working on making the FM reactive ;). I'm also extremely unfamiliar with the whole UI layout/widget system, so I can't even tell you how much work that'd really be).

Yeah, me too.

On the other side having a reactive FM that can be w > h without rotating itself could be useful for other platforms as well.

**Comment by NiLuJe** [comment]
> On the other side having a reactive FM that can be w > h without rotating itself could be useful for other platforms as well.

Agreed ;).

**Comment by poire-z** [comment]
>  My suggestion here would be NOT to remember the previous rotation of the book and always open the book in the current screen orientation.

If I understand correctly, @SilverGreen93 may not want FM in landscape, but just either in "portrait" or "inverted_portrait"?

> why would you put them back on the right in the FM? (Because the FM always spawns in Portrait w/ buttons on the right, maybe?

Might be the root issue indeed: it would switch from inverted_portrait to portrait:
https://github.com/koreader/koreader/blob/b111ccc2b803fc78425d9791fcc8bb7a9a704e6e/frontend/apps/filemanager/filemanager.lua#L49-L54
https://github.com/koreader/koreader/blob/b111ccc2b803fc78425d9791fcc8bb7a9a704e6e/frontend/apps/filemanager/filemanager.lua#L992-L996

('Reminder: switching a book from portait to inverted_portrait costs nothing. But switching it from portrait to landscape because of gyro-detected rotation would be unwelcome: for EPUBs, crengine would re-render the whole book, which may hang the device for dozens of seconds.)

**Comment by SilverGreen93** [comment]
@poire-z Yes, I don't talk about landscape here. I'm interested in portrait vs inverse-portrait here.

@NiLuJe Well, the real-life scneario is as follows: I read a book in portrait mode, then I want to change the hand that holds the eReader, so I rotate display 180 degrees in inverse-portrait mode, then quit the book and revert to file manger. Then lock the eReader. Then open the ereader in portrait mode. Then open the book, and the book now is in inverse-portrait mode which is annoying. And as I said this requires a full 360 degrees turn and wait 2 times to register rotation...

**Comment by SilverGreen93** [comment]
Maybe we could do something like:
- remember screen orientation if it was landscape
- else do not remember inverse-portrait, just remember portrait and when loading the book, if current orientation is not landscape, follow the current orientation of the display?

**Comment by NiLuJe** [comment]
(Probably won't have time to look into this for, err, a while. Not forgetting about it, it's just, like, way, way down below the priority list ;)).

**Comment by NiLuJe** [comment]
To be clear: not honoring the in-book stored orientation is non-negotiable, that's never going to change.

I'm only concerned about the "going back to FM" interaction.

**Comment by NiLuJe** [comment]
Ooh, it's sneakier than I thought.

Say, your default reading orientation is Portrait (i.e., the default).
You switch to reading a book in Inverted Portrait.
You go back to the FM. FM is still in Inverted Portrait (yay).
You re-open that book. It rotates to Portrait o_O.
(Same thing happens if you open a book set to Portrait, though. No issue with Landscape/Inverted Landscape).

**Comment by NiLuJe** [comment]
The (main) issue was actually that a gyro swap from portrait to Portrait wasn't reflected in the book's sidecar, which happily kept on assuming the book ought to be shown in Portrait.
